### PR TITLE
fix existingClaim not use

### PR DESCRIPTION
**Pull request by htlhenry**
#### What this PR does / why we need it:
The `persistence.existingClaim` in rabbitmq chart is not used, this PR will fixes it.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

**Comment by tompizmor** [comment]
Hi @htlhenry,

The rabbitmq chart didn't support providing an existing claim so I think there is nothing to fix.  

In case you want to add the functionality yourself, please change the title of the PR. Apart from the change you suggested, you also need to implement the following changes:

- Update the minor version of the chart in the Chart.yaml file.
- Document the new property in the README
- Add the property to the values.yaml
- Add the property to the values-production.yaml


Also, you need to sign your commits. Please follow the [instructions provided](https://github.com/helm/charts/pull/12894/checks?check_run_id=95741352) by the DCO bot

**Comment by htlhenry** [comment]
HI @tompizmor,
I follow your  suggestion  and make some change, please review again.

I see that this document already has one description about `persistence.existingClaim`，i think that  is not in use so i want to fixes, but may be a functionality :)

File: stable/rabbitmq/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: rabbitmq
-version: 4.11.1
+version: 4.12.1
```
Comment (tompizmor) [inline stable/rabbitmq/Chart.yaml]: The current version is `5.1.0` so you need to update to `5.2.0`
```suggestion
version: 5.2.0
```

**Review by tompizmor — CHANGES_REQUESTED** [review]

**Comment by tompizmor** [comment]
You can see that the `ci/circleci: lint-charts` check is failing because of the wrong chart version:

https://circleci.com/gh/helm/charts/67010?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by htlhenry** [comment]
I updated my chart version, but still not pass `ci/circleci: lint-charts`, i need some help.

File: stable/rabbitmq/templates/statefulset.yaml [hunk 2]
```diff
@@ -260,6 +260,10 @@ spec:
       {{- if not .Values.persistence.enabled }}
         - name: data
           emptyDir: {}
+      {{- else if (ne "" .Values.persistence.existingClaim) }}
```
Comment (carrodher) [inline stable/rabbitmq/templates/statefulset.yaml]:
```suggestion
      {{- else if .Values.persistence.existingClaim }}
```

File: stable/rabbitmq/templates/statefulset.yaml [hunk 3]
```diff
@@ -260,6 +260,10 @@ spec:
...
+        - name: data
+          persistentVolumeClaim:
+            claimName: {{ .Values.persistence.existingClaim }}
```
Comment (carrodher) [inline stable/rabbitmq/templates/statefulset.yaml:270]:
```suggestion
            claimName: {{ .Values.persistence.existingClaim }}
        {{- end }}
```

**Review by carrodher — CHANGES_REQUESTED** [review]
There are some errors in the yaml. I didn't test my suggestions so use it as pseudo-code, but the solution should be something similar.

**Review by tompizmor — APPROVED** [review]

**Comment by carrodher** [comment]
Hi, at this time tests are failing because the chart version is not ok, you need to bump the version.

File: stable/rabbitmq/Chart.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: rabbitmq
-version: 5.1.0
+version: 5.2.0
```
Comment (carrodher) [inline stable/rabbitmq/Chart.yaml]:
```suggestion
version: 5.3.0
```

Comment (carrodher) [reply stable/rabbitmq/Chart.yaml]: https://github.com/helm/charts/blob/master/stable/rabbitmq/Chart.yaml#L3

Comment (htlhenry) [reply stable/rabbitmq/Chart.yaml]: Ok, i updated the version.

**Review by tompizmor — APPROVED** [review]
/lgtm

**Pull request merged**